website/src/parsers/mdx/mdxhast.js

```diff
@@ -0,0 +1,56 @@
+import defaultParserInterface from '../utils/defaultParserInterface';
+import pkg from '@mdx-js/mdx/package.json';
+import mdxAstToMdxHast from '@mdx-js/mdx/mdx-ast-to-mdx-hast'
+
+const ID = 'MDXHAST';
+
+function removeNewlines(node) {
+  if (node.children != null) {
+    node.children = node.children.filter(node => node.value !== '\n');
+    node.children.forEach(removeNewlines);
+  }
+}
+
+export default {
+  ...defaultParserInterface,
+
+  id: ID,
+  displayName: ID,
+  version: pkg.version,
+  homepage: 'https://mdxjs.com',
+  locationProps: new Set(['position']),
+
+  loadParser(callback) {
+    require(['@mdx-js/mdx'], callback);
+  },
+
+  parse(mdx, code) {
+    let result;
+
+    mdx.sync(code, {
+      hastPlugins: [
+        mdxAstToMdxHast,
+        () => removeNewlines,
```

Comment:
This is only here to reduce noise. Each blank line turns into a text node and it's really noisy to read, I don't think it's useful at all. So I recursively remove all text nodes containing `\n`.